plugins/uefi-capsule/make-images.py

```diff
@@ -6,177 +6,153 @@
-                if ink.height == 0 or ink.width == 0:
-                    return False
-                data.update({"log": log, "ink": ink})
-                return True
-            data = {}
-            fs.foreach(find_size, data)
-
-            if len(data) == 0:
-                print("Missing sans fonts")
-                sys.exit(2)
-            log = data['log']
-            ink = data['ink']
-
-            height = math.ceil(max(ink.height, log.height) / Pango.SCALE)
-            width = math.ceil(max(ink.width, log.width) / Pango.SCALE)
-
-            x = -math.ceil(log.x / Pango.SCALE)
-            y = -math.ceil(log.y / Pango.SCALE)
-
-            img = cairo.ImageSurface(cairo.FORMAT_RGB24, width, height)
-            cctx = cairo.Context(img)
-            layout = PangoCairo.create_layout(cctx)
-            pctx = layout.get_context()
-            PangoCairo.context_set_font_options(pctx, font_option)
-
-            cctx.set_source_rgb(1, 1, 1)
-            cctx.move_to(x, y)
-            def do_write(fs, f, data):
-                """ write out glyphs """
-                ink = gs.extents(f)[0]
-                if ink.height == 0 or ink.width == 0:
-                    return False
-                PangoCairo.show_glyph_string(cctx, f, gs)
-                return True
-            fs.foreach(do_write, None)
-            img.flush()
-            try:
-                tmpname = filename[len(os.environ['DESTDIR']):]
-            except KeyError:
-                tmpname = filename[len(os.environ['MESON_BUILD_ROOT']):]
-            print("Writing {}".format(tmpname))
-            def make_dir(d):
-                """ make our parent dir and then our own dir """
-                if os.access(d, os.O_RDWR):
-                    return
-                parts = os.path.split(d)
-                if parts and parts[0]:
-                    make_dir(parts[0])
-                os.mkdir(d, 0o755)
-            d = os.path.split(filename)[0]
-            make_dir(d)
-
-            # write PNG
-            with io.BytesIO() as io_png:
-                img.write_to_png(io_png)
-                io_png.seek(0)
-
-                # convert to BMP
-                with io.BytesIO() as io_bmp:
-                    pimg = Image.open(io_png)
-                    pimg.save(io_bmp, format='BMP')
-                    io_bmp.seek(0)
-                    with gzip.open(filename, 'wb') as f:
-                        f.write(io_bmp.read())
-
-        for lang in self.languages:
-            #print("lang:\"%s\" string:\"%s\"" % (lang, string))
+
+def languages(linguas):
+    """ Find all of our languages """
+    for x in open(linguas, "r").readlines():
+        x = x.strip()
+        # currently ligatures seem completely broken in hi
+        if x == "hi":
+            continue
+        yield x
+    yield "en"
+
+
+def main(args) -> int:
+
+    domain = "fwupd"
+    gettext.textdomain(domain)
+    gettext.install(domain, args.localedir)
```

Comment:
do you still need this?

Reply:
Urgh, yes, and no. If you ninja install then run the script then the .mo files exist and it works. From a clean checkout this gets run before the .mo files get created. From a 40,000ft view its insane we're creating a BMP file from a .mo file from a .po file, and I think we can skip out the middle step.